### PR TITLE
Version based biomes

### LINKED ISSUE #569: Biome data values
Hello the data values of certain biomes have been changed since 1.13 so could these be updated in the GUI aswell. Or add a legacy toggle for the old names?

### PR DESCRIPTION
**Pull request by burgerguy**
Basically, the `Biome` objects are no longer static. The list of biomes is now in a VersionFeature. The biome index has been renamed to id, and the recent nether biomes have been added. Also, the default biome profile now just uses a file built straight into amidst, meaning it cannot be removed when it is compiled. Finally, all of the things that use lists of biomes for validating structures have been changed to lists of integers. There is still a bunch of static ints in the Biome class named to their biome counterparts for specific use cases.

If you really need to use the full list of biomes from the latest version, you can use:
```java
DefaultVersionFeatures.builder(null, null).create(RecognisedVersion.UNKNOWN).get(FeatureKey.BIOME_LIST);
```
However it isn't recommended that it should be used outside of very specific cases (ex. biome profiles) and testing purposes. Originally this was implemented in to replace Biome.allBiomes() method, but I wanted to discourage it's usage. Also it screwed up some tests due to the order of the classes loading. I didn't really look into it as I was happy to remove it.

This is probably my biggest PR yet, so let me know if you catch any bugs I missed or want to see any changes to it.

Edit: I also forgot to say I added guava to the dependencies initially for a BiMap implementation, but I ended up ditching that idea. I'm now using it for getting the default biome profile.

Edit 2: this fixes #569

### CODE REVIEW

**Comment by moulins** [comment]
Nice, but this breaks any custom biome profile currently in use; it would be better to still support the old string keys with the old names (maybe with a warning in the logs?).

File: src/main/java/amidst/settings/biomeprofile/BiomeProfile.java [hunk 1]
```diff
@@ -2,52 +2,61 @@
...
 @Immutable
 public class BiomeProfile {
-	private static Map<String, BiomeColorJson> createDefaultColorMap() {
-		Map<String, BiomeColorJson> result = new HashMap<>();
-		for (Biome biome : Biome.allBiomes()) {
-			result.put(biome.getName(), biome.getDefaultColor().createBiomeColorJson());
+	private static BiomeProfile createDefaultProfile() {
+		final String profileFile = "/amidst/mojangapi/default_biome_profile.json";
+		try (InputStream stream = BiomeProfile.class.getResourceAsStream(profileFile)) {
+			return JsonReader.readString(CharStreams.toString(new InputStreamReader(stream, Charsets.UTF_8)), BiomeProfile.class);
```
[the remark this answers is not in the document]
Comment (burgerguy) [reply src/main/java/amidst/settings/biomeprofile/BiomeProfile.java]: Fixed this in latest push and removed Guava.

File: src/main/java/amidst/mojangapi/world/biome/UnknownBiomeNameException.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package amidst.mojangapi.world.biome;
+
+import amidst.documentation.Immutable;
+
+@SuppressWarnings("serial")
```
[the remark this answers is not in the document]
Comment (burgerguy) [reply src/main/java/amidst/mojangapi/world/biome/UnknownBiomeNameException.java]: Also fixed this in latest push.

**Comment by burgerguy** [comment]
@moulins I just pushed the changes for automatic conversion of old biome profiles to the new format, so it shouldn't be a problem anymore.

**Comment by moulins** [comment]
> @moulins I just pushed the changes for automatic conversion of old biome profiles to the new format, so it shouldn't be a problem anymore.

Perfect! I'll try to test everything and merge tomorrow. ;)

**Comment by moulins** [comment]
I've done some tests, and everything works. Thanks for the hard work!

However, I've noticed that the default profile isn't automatically saved to the `biome` directory (if it exists) anymore. This is problematic, because having the default profile automatically saved gave the user an "example file" to modify when creating their own biome profiles. Without this file, the user can't create new biome profiles easily.

**Comment by burgerguy** [comment]
The default biome profile currently exists inside of the resources folder. We have to keep it there so we know it's secure from accidental deletion/changes if someone were to modify the biome profiles. This used to not be as important, but now that the biome objects don't have default biome colors, we have to rely on it more than ever. A better idea may be to have a button in the biome profiles menu that makes a copy of the default profile and put it in the biome folder so it can be modified.

**Comment by burgerguy** [comment]
There now is a button under "Reload Biome Profiles" that allows you to create an example biome profile to edit. Before, users wouldn't be able to easily tell that there was an option for a shortcut with biome profiles (I actually didn't know this was an option). Now when creating an example profile, it sets the shortcut to Ctrl + E. I also set the default biome profile shortcut to Ctrl+D.

**Comment by moulins** [comment]
> There now is a button under "Reload Biome Profiles" that allows you to create an example biome profile to edit.

This is indeed better than what we had before, thanks!

**Comment by machinesmith42** [comment]
will this be in the next release?

**Comment by moulins** [comment]
Yes, this will be included in the next version of Amidst. However, note that there isn't any planned date for the next release for now.

**Pull request merged**